Content.Client/BarSign/Ui/BarSignMenu.xaml.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Linq;
+using Content.Client.UserInterface.Controls;
+using Content.Shared.BarSign;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
+
+namespace Content.Client.BarSign.Ui;
+
+[GenerateTypedNameReferences]
+public sealed partial class BarSignMenu : FancyWindow
+{
+    private readonly IPrototypeManager _prototype;
+
+    private string? _currentId;
+
+    private readonly List<BarSignPrototype> _cachedPrototypes = new();
+
+    public event Action<string>? OnSignSelected;
+
+    public BarSignMenu(EntityUid owner, IEntityManager entMan, IPrototypeManager protoMan)
+    {
+        RobustXamlLoader.Load(this);
+        _prototype = protoMan;
+        _currentId = entMan.GetComponentOrNull<BarSignComponent>(owner)?.Current;
+
+        _cachedPrototypes.Clear();
+        _cachedPrototypes = Shared.BarSign.BarSignSystem.GetAllBarSigns(_prototype)
+            .OrderBy(p => Loc.GetString(p.Name))
+            .ToList();
```

Comment:
uhhh, this is good too but i meant getting the Current bar sign in the BUI so you dont have to pass the owner (prob pass the prototype list as well if prot reloading is chill with that)